src/plugins/interactive_setup/server/routes/enroll.ts

```diff
@@ -6,26 +6,143 @@
  * Side Public License, v 1.
  */
 
+import { constants } from 'fs';
+import fs from 'fs/promises';
+import path from 'path';
+
 import { schema } from '@kbn/config-schema';
 
 import type { RouteDefinitionParams } from './';
 
 /**
  * Defines routes to deal with Elasticsearch `enroll_kibana` APIs.
  */
-export function defineEnrollRoutes({ router }: RouteDefinitionParams) {
+export function defineEnrollRoutes({
```

Comment:
__question:__ the `enroll` and `configure` routes looks very similar, what is the difference between these two?

__note:__ I've also noticed that when I use manual configuration and try to connect to HTTP address with disabled security in ES, Kibana still tries to call `_authenticate` method, but this endpoint doesn't exist in this case. I haven't looked at the flow yet, but for manual configuration mode we may need to first check if authentication is enabled (e.g. check that ping doesn't fail with 401) before we call `_authenticate`.

Reply:
> question: the enroll and configure routes looks very similar, what is the difference between these two?

Enroll endpoint uses the ES enrolment API to generate kibana system credentials for you. The CA comes from the ES enrollment response. 

Configure endpoint allows you to pass in the actual configuration values themselves including the CA which will be written to config as is. 

There's a bit of shared logic but nothing that can't be refactored. Both of them need to verify connectivity and write the configuration to disc but their slightly different endpoints I think. We could definitely consolidate them if you prefer though. Is there a specific concern you'd like addressed? 

> note: I've also noticed that when I use manual configuration and try to connect to HTTP address with disabled security in ES, Kibana still tries to call _authenticate method, but this endpoint doesn't exist in this case. I haven't looked at the flow yet, but for manual configuration mode we may need to first check if authentication is enabled (e.g. check that ping doesn't fail with 401) before we call _authenticate.

Good catch, I'll add a note to fix this.

Reply:
> There's a bit of shared logic but nothing that can't be refactored. Both of them need to verify connectivity and write the configuration to disc but their slightly different endpoints I think. We could definitely consolidate them if you prefer though. Is there a specific concern you'd like addressed?

Not really a concern, just wondering if it'd make sense to move any non-trivial logic to a shared place and test it once and separately from the route logic itself.